backend/data/handlers/calendar.js

```diff
@@ -0,0 +1,176 @@
+const util = require('./util');
+const { OAuth2Client } = require('google-auth-library');
+const authConfig = require('./auth.config.json');
+const {google} = require('googleapis');
+
+const calendar = {};
+
+const REQUIRED_PARAMS = {
+    ADD_EVENT: ['startTime', 'endTime', 'title'],
+    UPDATE_EVENT: ['eventId']
+}
+const _hasRequiredParameters = (requiredParams, paramsToCheck, res) => {
+    if (!requiredParams.every(p => paramsToCheck.hasOwnProperty(p))) {
+        res.statusCode = 400;
+        throw Error("Missing one or more of these required fields: {" + requiredParams.join(", ") + "}");
+    }
+}
+
+const _checkDate = (date, res) => {
+    if (!(date instanceof Date) || isNaN(date)) {
+        res.statusCode = 400;
+        throw Error("Invalid Date provided");
+    }
+}
+
+/**
+ * Add a new Google Calendar Event
+ * 
+ * @param {String} token: access_token for Oauth
+ * @param {Object} eventDetails: contains details about the calendar event
+ * 
+ * @returns Details of the Calendar Event that got added
+ */
+calendar.add = async (token, eventDetails, res) => {
+    _hasRequiredParameters(REQUIRED_PARAMS.ADD_EVENT, eventDetails, res);
+    const eventStartTime = new Date(eventDetails['startTime']);
+    const eventEndTime = new Date(eventDetails['endTime']);
+    _checkDate(eventStartTime, res);
+    _checkDate(eventEndTime, res);
+
+    const client = new OAuth2Client(authConfig['client_id']);
+    client.setCredentials({
+        access_token: token
+    });
+    const calendar = google.calendar({version: 'v3', auth: client});
```

Comment:
Shadow naming used here. It's not a good idea to reuse an identifier from a parent scope (calendar in this case) in the child scope because it can lead to confusion as to which version of calendar you are referring to.